File: codes/OOP/java/Interfaces.java
/**
 * Created by gaurav on 6/2/16.
 */
interface Shape {
    int getHeight();
    int getWidth();
    void setHeight(int h);
    void setWidth(int w);
}
class Rectangle implements Shape{
    int height, width;

    @Override
    public int getHeight() {
        return height;
    }

    @Override
    public int getWidth() {
        return width;
    }

    @Override
    public void setHeight(int h) {
        height = h;
    }

    @Override
    public void setWidth(int w) {
        width = w;
    }
}
public class Interfaces {
    public static void main(String[] args) {
        Rectangle rectangle = new Rectangle();
        rectangle.setHeight(5);
        rectangle.setWidth(4);
        System.out.println(rectangle.getHeight());
        System.out.println(rectangle.getWidth());
    }
}

File: codes/OOP/cpp/interfaces.cpp
//An interface describes the behavior or capabilities of a C++ class without committing to a particular implementation of that class.
#include<bits/stdc++.h>
using namespace std;

class Shape {
protected:
    int width, height;
public:
    virtual void print_area() = 0;
    void set_width(int w) {
        width = w;
    }
    void set_height(int h) {
        height = h;
    }
};

class Rectangle: public Shape {
public:
    void print_area() {
        cout << width*height << endl;
    }
};

class Triangle: public Shape {
public:
    void print_area() {
        cout << (width*height)/2 << endl;
    }
};

int main( )
{
    Rectangle rect;
    rect.set_height(5);
    rect.set_width(8);
    rect.print_area();
   return 0;
}
